src/layers/linear.rs

```diff
@@ -29,4 +35,32 @@ impl Layer<Ix1, Ix1> for Linear {
     fn forward(&self, input: &Tensor<Ix1>) -> Tensor<Ix1> {
         input.dot(&self.weights) + &self.biases
     }
+
+    fn weights(&self) -> Vec<f64> {
+        self.weights.iter().map(|v| v.value()).collect()
+    }
+
+    fn biases(&self) -> Vec<f64> {
+        self.biases.iter().map(|v| v.value()).collect()
+    }
+
+    fn set_weights(&mut self, new_weights: Vec<f64>) {
+        for (v, weight) in self.weights.iter_mut().zip(new_weights) {
+            *v = weight.into();
+        }
+    }
+
+    fn set_biases(&mut self, new_biases: Vec<f64>) {
+        for (v, bias) in self.biases.iter_mut().zip(new_biases) {
+            *v = bias.into();
+        }
+    }
+
+    fn is_trainable(&self) -> bool {
+        true
+    }
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
```

Comment:
I really wish these were able to be associated consts but doing so won't allow this trait to become a trait object :(